packages/components/src/Tree/Tree.test.tsx

```diff
@@ -124,6 +128,38 @@ describe('Tree', () => {
     expect(onClose).toHaveBeenCalledTimes(0)
   })
 
+  test('Key presses on detail do not open the Tree or trigger callbacks when accessory === true', () => {
+    const onOpen = jest.fn()
+    const onClose = jest.fn()
+    const { getByText, queryByText } = renderWithTheme(
+      <Tree
+        label="Tree Label"
+        detail={{
+          content: <Button>Tree Detail</Button>,
+          options: { accessory: true },
+        }}
+        onClose={onClose}
+        onOpen={onOpen}
+      >
+        Hello World
+      </Tree>
+    )
+
+    const detail = getByText('Tree Detail')
+
+    expect(queryByText('Hello World')).not.toBeInTheDocument()
+    fireEvent.keyDown(detail, {
+      key: 'Enter',
+    })
+    expect(queryByText('Hello World')).not.toBeInTheDocument()
+    expect(onOpen).toHaveBeenCalledTimes(0)
+    fireEvent.keyDown(detail, {
+      key: 'Enter',
+    })
+    expect(queryByText('Hello World')).not.toBeInTheDocument()
+    expect(onClose).toHaveBeenCalledTimes(0)
```

Comment:
Do you actually need multiple Enter key presses here? Seems like this would work just as well?

```suggestion
    fireEvent.keyDown(detail, {
      key: 'Enter',
    })
    expect(queryByText('Hello World')).not.toBeInTheDocument()
    expect(onOpen).toHaveBeenCalledTimes(0)
    expect(onClose).toHaveBeenCalledTimes(0)
```

Reply:
We probably don't. The intent is to make sure that both `onOpen` and `onClose` don't trigger (i.e. on the first and second click of a closed `Tree`, respectively), but if the `Tree` "opening" its accordion on the first clicks it's probably safe to assume that it's not "closing" on the second one / firing it's `onClose`.

I do this in `Clicks on detail do not open the Tree or trigger callbacks when accessory === true` as well and I think it's correct to remove the extra click / keypress there as well.

Resolved in 4e225d48d